### PR TITLE
Added log to show which URL is blocked by CORS.

### PR DESCRIPTION
**Pull request by Aryakoste**
#### Summary

Added log which shows which url is blocked by Cross Origin.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/26235
Jira https://mattermost.atlassian.net/browse/MM-56662

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/channels/api4/websocket.go [hunk 1]
```diff
@@ -30,6 +30,10 @@ func connectWebSocket(c *Context, w http.ResponseWriter, r *http.Request) {
 		CheckOrigin:     c.App.OriginChecker(),
 	}
 
+	if !upgrader.CheckOrigin(r) {
+		c.Logger.Warn("URL Blocked Because of CORS", mlog.String("Blocked Origin", r.Header.Get("Origin")))
+	}
+
```
Comment (hanzei) [inline server/channels/api4/websocket.go:36]: The change is fine, but the downside of how it's implemented is that `upgrader.CheckOrigin` gets called twice. I wonder if we can get the log message into the checker method. Is it an option to pass a logger to `c.App.OriginChecker` and log the warning in there?

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the PR @Aryakoste :tada:

**Review by hanzei — APPROVED** [review]
Nice work :+1:

**Review by agnivade — CHANGES_REQUESTED** [review]
I think this would be the wrong place to add the URL. The better place is to add the origin to the original log message which shows the failure, rather than splitting the info in two places.

Please see api4/websocket.go:L35, and then add a new param with the `r.Header["Origin"]`, and also modify the en.json accordingly to log the new param.

File: server/channels/app/server.go [hunk 2]
```diff
@@ -1114,7 +1114,7 @@ func (s *Server) stopLocalModeServer() {
 	}
 }
 
-func (a *App) OriginChecker() func(*http.Request) bool {
+func (a *App) OriginChecker(logger *mlog.Logger) func(*http.Request) bool {
```
Comment (agnivade) [inline server/channels/app/server.go:1117]: We can revert this now.

File: server/channels/app/server.go [hunk 3]
```diff
@@ -1149,6 +1149,7 @@ func (a *App) OriginChecker() func(*http.Request) bool {
 		if err != nil {
 			return false
 		}
+
```
Comment (agnivade) [inline server/channels/app/server.go:1152]: Unnecessary newline

**Review by agnivade — COMMENTED** [review]
Looking good! Just some small things to take care of.

File: server/channels/app/server_test.go [hunk 4]
```diff
@@ -571,7 +571,7 @@ func TestOriginChecker(t *testing.T) {
 		r := &http.Request{
 			Header: http.Header{"Origin": []string{fmt.Sprintf("%s%s", tc.HeaderScheme, tc.HeaderHost)}},
 		}
-		res := th.App.OriginChecker()(r)
+		res := th.App.OriginChecker(th.TestLogger)(r)
```
Comment (agnivade) [inline server/channels/app/server_test.go:574]: This will need to be removed as well.

**Comment by hanzei** [comment]
@Aryakoste heads up that `TestWebSocketUpgrade` is failing due to your changes. Please update the tets.

**Comment by Aryakoste** [comment]
Done

**Review by agnivade — APPROVED** [review]
Thanks!

**Pull request merged**